components/server/ee/src/billing/billing-mode.spec.db.ts

```diff
@@ -0,0 +1,548 @@
+                config: {
+                    enablePayment: false,
+                    usageBasedPricingEnabled: true,
+                },
+                expectation: {
+                    mode: "none",
+                },
+            },
+            {
+                name: "payment disabled (ubb: false)",
+                subject: user(),
+                config: {
+                    enablePayment: false,
+                    usageBasedPricingEnabled: false,
+                },
+                expectation: {
+                    mode: "none",
+                },
+            },
+            {
+                name: "payment enabled (ubb: false)",
+                subject: user(),
+                config: {
+                    enablePayment: true,
+                    usageBasedPricingEnabled: false,
+                },
+                expectation: {
+                    mode: "chargebee",
+                    tier: "free",
+                },
+            },
+            // user: chargebee
+            {
+                name: "user: chargbee paid personal",
+                subject: user(),
+                config: {
+                    enablePayment: true,
+                    usageBasedPricingEnabled: false,
+                    subscriptions: [subscription(Plans.PERSONAL_EUR)],
+                },
+                expectation: {
+                    mode: "chargebee",
+                    tier: "paid",
+                    planIds: [Plans.PERSONAL_EUR.chargebeeId],
+                    hasPersonalPlan: true,
+                },
+            },
+            {
+                name: "user: chargbee paid team seat",
+                subject: user(),
+                config: {
+                    enablePayment: true,
+                    usageBasedPricingEnabled: false,
+                    subscriptions: [subscription(Plans.TEAM_PROFESSIONAL_EUR)],
+                },
+                expectation: {
+                    mode: "chargebee",
+                    tier: "paid",
+                    planIds: [Plans.TEAM_PROFESSIONAL_EUR.chargebeeId],
+                    hasPersonalPlan: false,
+                },
+            },
+            {
+                name: "user: chargbee paid personal + team seat",
+                subject: user(),
+                config: {
+                    enablePayment: true,
+                    usageBasedPricingEnabled: false,
+                    subscriptions: [subscription(Plans.PERSONAL_EUR), subscription(Plans.TEAM_PROFESSIONAL_EUR)],
+                },
+                expectation: {
+                    mode: "chargebee",
+                    tier: "paid",
+                    planIds: [Plans.PERSONAL_EUR.chargebeeId, Plans.TEAM_PROFESSIONAL_EUR.chargebeeId],
+                    hasPersonalPlan: true,
+                },
+            },
+            // user: transition chargebee -> UBB
+            {
+                name: "user: chargbee paid personal (cancelled) + team seat",
```

Comment:
- cancellationDate is the date when we click "cancel"
- endDate is when the subscription is no longer effective (currently: end of billing period)

Instead of setting endDate to "end of billing period", I think should prorate and set it to "cancellationDate + 24h", to give room for a safe transition period from Chargebee to Stripe (as discussed [here internally](https://www.notion.so/gitpod/Usage-Based-Pricing-Checkin-1185a6c70c1f4a93887b72cf5d225509#e196401e536a440aa91a804facad3280)).

:cloud: I'm just realizing that the current implementation here does not support that use-case very well, though. :cry: 
If you are still on a team seat, and deliberately want to upgrade to "personal usage-based", then you also want to ripe the benefits of that. @jankeromnes Would be awesome if we can find 10mins of sync time on this and adjust.

Reply:
Addressed [here](https://github.com/gitpod-io/gitpod/pull/11812#issuecomment-1205029906).